File: src/main/java/com/github/sorusclient/client/transform/Result.java
package com.github.sorusclient.client.transform;

import java.util.Collections;
import java.util.List;
import java.util.function.Consumer;

public class Result<T> {

    private final List<T> results;

    public Result(List<T> results) {
        this.results = results;
    }

    public void apply(Consumer<? extends T> consumer) {
        for (T result : this.results) {
            ((Consumer<T>) consumer).accept(result);
        }
    }

    public Result<T> nth(int index) {
        return new Result<>(Collections.singletonList(this.results.get(index)));
    }

}

File: src/main/kotlin/com/github/sorusclient/client/transform/Result.kt
package com.github.sorusclient.client.transform

import java.util.function.Consumer

class Result<T>(private val results: List<T>) {

    fun apply(consumer: Consumer<out T>) {
        for (result in results) {
            (consumer as Consumer<T>).accept(result)
        }
    }

    fun nth(index: Int): Result<T> {
        return Result(listOf(results[index]))
    }

    fun nth(indices: IntRange)  {
        Result(listOf(results.subList(indices.first, indices.last)))
    }

}